### PR TITLE
[KUBOS-506] [KUBOS-507] Docs for libcsp & freertos

### PR DESCRIPTION
**Pull request by plauche**
- [x] Migrating freertos to sphinx
- [x] Migrating libcsp to sphinx

### CODE REVIEW

File: docs/libcsp/csp_docs/example.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
+Client and server example
```
Comment (kyleparrott) [inline docs/libcsp/csp_docs/example.rst]: "server" and "example" should be capitalized

File: docs/libcsp/csp_docs/history.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+History
+=======
+
+The idea was developed by a group of students from Aalborg University in 2008. In 2009 the main developer started working for GomSpace, and CSP became integrated into the GomSpace products. The protocol is based on a 32-bit header containing both transport, network and MAC-layer information. It's implementation is designed for, but not limited to, embedded systems such as the 8-bit AVR microprocessor and the 32-bit ARM and AVR from Atmel. The implementation is written in C and is currently ported to run on FreeRTOS and POSIX and pthreads based operating systems like linux and bsd. The three letter acronym CSP was originally an abbreviation for CAN Space Protocol because the first MAC-layer driver was written for CAN-bus. Now the physical layer has extended to include spacelink, I2C and RS232, the name was therefore extended to the more general Cubesat Space Protocol without changing the abbreviation.
+
+Satellites using CSP
+--------------------
+
+This is the known list of satellites or organisations that uses CSP.
```
Comment (CatFreed) [inline docs/libcsp/csp_docs/history.rst]: nit: "organizations"

**Comment by plauche** [comment]
[Docs: freertos API](https://trello.com/c/CIY7c8ul/18-docs-freertos-api)

**Comment by plauche** [comment]
[Docs: CSP API](https://trello.com/c/xJOOpuBb/20-docs-csp-api)

**Pull request merged**